### PR TITLE
[CI] Migrate to 1ES

### PR DESCRIPTION
**Pull request by DmitriyKirakosyan**
- Migrating Pipelines to 1ES Template
- Add APIScan

[AB#104414](https://dev.azure.com/msmobilecenter/Mobile-Center/_boards/board/t/Mobile%20SDK%20team/Backlog%20Items/?workitem=104414)

Build - https://dev.azure.com/msmobilecenter/SDK/_build/results?buildId=1518658

### CODE REVIEW

File: .azurepipelines/build-release-packages-1es.yml [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+          inputs:
+            targetType: filePath
+            filePath: ./build.ps1
+            arguments: '"version.cake" --Target="SetReleaseVersion"'
+
+        - powershell: |
+            Set-Location "C:\Program Files (x86)\Microsoft Visual Studio\Installer\"
+            $InstallPath = "C:\Program Files\Microsoft Visual Studio\2022\Enterprise"
+            $componentsToAdd = @(
+                "Microsoft .NET Framework 4.6.1 Targeting Pack"
```
Comment (lucen-ms) [inline .azurepipelines/build-release-packages-1es.yml:65]: Why do we need .net 4.6.1?

File: .azurepipelines/build-release-packages-1es.yml [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+        - task: cake-build.cake.cake-build-task.Cake@2
+          displayName: 'Prepare DotNet Assemblies'
+          inputs:
+            target: PrepareAssemblies
+            arguments: '--StorageAuthParams="$(appcenter-sdk-blob-storage-access-secret)"'
+            Version: 2.2.0
+
+        - task: UseDotNet@2
+          displayName: 'Install .NET 3.1'
```
Comment (lucen-ms) [inline .azurepipelines/build-release-packages-1es.yml:136]: Why do we need .net 3.1?

File: .azurepipelines/build-release-packages-1es.yml [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+              "parameters": [],
+              "toolName": "signtool.exe",
+              "toolVersion": "6.2.9304.0"
+              }
+              ]
+            SessionTimeout: 45
+            VerboseLogin: true
+
+        - task: CopyFiles@2
+          displayName: 'Move assemblies to bin'
```
[the remark this answers is not in the document]
Comment (DmitriyKirakosyan) [reply .azurepipelines/build-release-packages-1es.yml]: fixed

File: .azurepipelines/build-release-packages-1es.yml (hunk 2 above)
Comment (DmitriyKirakosyan) [reply .azurepipelines/build-release-packages-1es.yml]: Removed. It was added as part of testing on newer macos image, for installing mono and msbuild. Preinstalled on the current image.

File: .azurepipelines/build-release-packages-1es.yml [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+          displayName: 'Set Release Version'
+          inputs:
+            targetType: filePath
+            filePath: ./build.sh
+            arguments: '"version.cake"  -t SetReleaseVersion'
+
+        - bash: 'sudo $AGENT_HOMEDIRECTORY/scripts/select-xamarin-sdk.sh 6_12_0'
+          displayName: 'Use Xamarin Bundle 6.12.0'
+
+        - task: JavaToolInstaller@0
```
Comment (lucen-ms) [inline .azurepipelines/build-release-packages-1es.yml:105]: What for do we use Java?

**Review by lucen-ms — APPROVED** [review]

File: .azurepipelines/build-release-packages-1es.yml (hunk 4 above)
Comment (DmitriyKirakosyan) [reply .azurepipelines/build-release-packages-1es.yml:105]: This specific version of java was used to build Xamarin projects for Android.

**Pull request merged**